Fix latest Coverity issues

jamessan opened the pull request:
coverity/161194: Restore check for 'keywordprg' being ":help"
coverity/161195: Increase scope of exe_name

jamessan committed:
coverity/161195: Increase scope of exe_name

Since exe_name is a stack allocated array, we need it to be in scope for
the lifetime that vim_path points to it.

jamessan committed:
coverity/161194: Restore check for 'keywordprg' being ":help"

998d0ffc09d5c7358db62dc88c2e2b87622f60b5 removed the explicit check for
":help", relying instead on whether the user was in a help buffer.
However, this breaks escaping the identifier for use in the lookup
command.

2f54d6927cc02484b528a5e8b25b64c8d6580ddd tried to fix this by removing
"!kp_ex" in "if (cmdchar == 'K' && !kp_ex)", but that causes shell
escaping to be used instead of escaping for tag lookup.

justinmk commented:
👍  Thanks :)

 For CID 148062 we can just add CheckIncludeFiles.c to the coverity "ignore" regex, I think.

jamessan commented:
> For CID 148062 we can just add CheckIncludeFiles.c to the coverity "ignore" regex, I think.

It's already ignored.  It just goes out in the email for some reason, but it doesn't show up on the dashboard.

jamessan committed:
ci: Update Coverity model for typval refactoring

[ci skip]

jamessan committed:
coverity/16127: Verify lang is non-NULL before calling strlen

jamessan committed:
coverity/161216: Ensure buf is valid for lifetime of defstr

Depending on the type of argument for input()/inputdialog()'s {text}
argument, defstr may point to buf.  Therefore it needs to be in scope
for the lifetime of defstr.

Also, use a different buffer for the handling of the 3rd argument to
input()/inputdialog().  Although the buffer defstr points to is used
immediately, it avoids potential mishaps if the code changes.

jamessan commented:
Off-hand I don't see a general way to model the ui_bridge functions.  We could add each function and note that there's an allocation escape, but it seems easier to just mark those in Coverity as intentional.

I'm done with this PR for now, pending any review comments.
